x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/plugin/ComputeService.java

```diff
@@ -289,12 +290,19 @@ private void startComputeOnDataNodes(
         ActionListener<Void> parentListener,
         Supplier<ActionListener<ComputeResponse>> dataNodeListenerSupplier
     ) {
+        var planWithReducer = configuration.pragmas().nodeLevelReduction() == false
```

Comment:
This is the meat of the PR. Adding the reducing topN, limit, sort (even though atm I don't think we can have such a situation, maybe with mv_expand...) only if the pragma allows it.